test/unit/host_test.rb

```diff
@@ -1123,6 +1123,19 @@ def teardown
       assert_kind_of Nic::Managed, host.interfaces.find_by_identifier('eth1')
     end
 
+    test "host can't have more interfaces with same identifier" do
+      host = FactoryGirl.build(:host, :managed)
+      host.primary_interface.identifier = 'eth0'
+      nic = host.interfaces.build(:identifier => 'eth0')
+      refute host.valid?
+      assert_present nic.errors[:identifier]
+      assert_present host.errors[:interfaces]
+      nic.identifier = 'eth1'
+      host.valid?
```

Comment:
this is on purpose, this test does not know whether the host will be valid? or not, there might be other validations added later, we just need to trigger all validations and then make sure that there are no errors for identifier and interfaces

note that in few lines above `refute` is also intentional, as we are pretty sure that we're making host invalid by setting bad identifier